### DISCUSSION TITLE
Opening the first file with Telescope

### DISCUSSION THREAD

**Discussion by danielcopper** [LazyVim/LazyVim#1376] [Q&A] [answered]
Hi everyone,

When i open the current directory with "nvim . " and then open a file through telescope, the "[No Name]" Buffer remains visible.
When i open the first file by selecting it from neotree this doesn't happen.

Is there a way to auto delete the no name Buffer when opening the first file with Telescope?

**Comment by dpetka2001** [comment]
This only happens to me when invoking Telescope while your cursor is in Neo-tree. Move your cursor to the "no name" buffer and then invoke Telescope. It should overwrite the "no name" buffer with the file you want to open from Telescope.

**Reply by danielcopper** [reply]
Yes that's a way to make it work. Thanks a lot!

Is there also a proper way to make neotree start in fullscreen?

Played around with it a bit but it lead to strange behaviour. For example if I close all other buffers except neotree, it shows in fullscreen and if i select a file from the tree is it openes like expected. So far so good.
But when i have Neotree in fullscreen and then open a file with Telescope, the file is opened, but Neotree appears on the right instead of left.

**Reply by dpetka2001** [reply]
`Neotree current` to make it fullscreen. I can't reproduce the behavior you're describing when Neotree is fullscreen. If I open a file with Telescope it just replaces the buffer that Neotree is opened in. Afterwards, if I press `<leader>e` to open the Neotree again it appears on the right like it's supposed to.

**Reply by dpetka2001** [reply]
One solution I could think of to make it start in fullscreen, would be to change in the Neotree `opts` to have the option `position = "current"`. But then you would have to create an explicit keybinding to make it open in sidebar, because with the above option it would always open in fullscreen.

**Reply by danielcopper** [reply]
Thanks a lot for all your input.

Somehow i can't get it to work like I want it to.
I tried running an autocmd like this:

`vim.api.nvim_create_autocmd("VimEnter", {
    command = "Neotree current" } )
`

That resulted in two instances of neotree. One on the left. On filling the rest of screen and the NoName buffer still there as well.
Also tried without current.
Set up two autocmds for VimEnter. One with Neotree, the second one to delete all other buffers. (don't know what the correct way is to chain multiple cmds in one autocmd).
Doing this manually works but not as autocmd.

Maybe its time to accept the state for now and tackle this at some later point...

**Reply by dpetka2001** [reply, marked as answer]
Try the following
```lua
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    keys = {
      {
        "<leader>e",
        function()
          require("neo-tree.command").execute({
            position = "left",
            toggle = true,
            dir = require("lazyvim.util").get_root(),
          })
        end,
        desc = "Open Neotree in sidebar",
      },
    },
    opts = {
      window = { position = "current" },
    },
  },
}
```
in a file inside `~/.config/lua/plugins/` directory. This way it will open full screen and if you want to open it in sidebar, you will have to use explicitly the keybinding. Give it a try to see if it meets your needs.

**Reply by danielcopper** [reply]
Yes!!

That actually works. Thanks a ton.
Just as a side note. There is a small difference to the way neo-tree normally opens in the current window/fullscreen.
The additional info like size type etc. is kept left aligned with the solution and normally spans the whole width of the window.

Still pretty awesome!

**Reply by dpetka2001** [reply]
From the photo on the Neo-tree [repo](https://github.com/nvim-neo-tree/neo-tree.nvim#neo-tree-filesystem-netrw-style) it seems that it is left aligned by default(or are you confused with netrw?). Maybe you should ask about that on the Neo-tree repo, cuz I don't know if there is an option about that.
